zypp/target/rpm/RpmDb.cc

```diff
@@ -1239,6 +1240,54 @@ namespace
 
     if ( ret != RpmDb::CHK_OK )
     {
+
+      bool didReadHeader = false;
+      std::unordered_map< std::string, std::string> fprs;
+
+      // we replace the data only if the key IDs are actually only 8 bytes
+      str::regex rxexpr( "key ID ([a-fA-F0-9]{8}):" );
+      for ( auto &detail : detail_r ) {
+        auto &line = detail.second;
+        str::smatch what;
+        if ( str::regex_match( line, what, rxexpr ) ) {
+
+          if ( !didReadHeader ) {
+            didReadHeader = true;
+
+            // Get signature info from the package header, RPM always prints only the 8 byte ID
+            auto header = RpmHeader::readPackage( path_r, RpmHeader::NOVERIFY );
+            if ( !header ) {
+              ERR << "Failed to read package" << std::endl;
+              return RpmDb::CHK_ERROR;
+            }
```

Comment:
If the header is for whatever reason not readable, I'd prefer to continue with the original workflow without replacing the FPR. Just leave the loop and return the computed value (ret).

Reply:
ok